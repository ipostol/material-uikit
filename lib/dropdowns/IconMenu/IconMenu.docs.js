'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.IconMenuDocs = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactMarkupComponents = require('react-markup-components');

var _More = require('./../../icons/More');

var _More2 = _interopRequireDefault(_More);

var _IconMenu = require('./IconMenu');

var _IconMenu2 = _interopRequireDefault(_IconMenu);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var options = [{
  value: 'Option1'
}, {
  value: 'Option2'
}];

var IconMenuDocs = exports.IconMenuDocs = function IconMenuDocs() {
  return _react2.default.createElement(
    _reactMarkupComponents.Skin,
    null,
    _react2.default.createElement(_IconMenu2.default, {
      options: options,
      onRight: true,
      icon: _react2.default.createElement(_More2.default, null)
    }),
    _react2.default.createElement('br', null),
    _react2.default.createElement('br', null)
  );
};

exports.default = IconMenuDocs;