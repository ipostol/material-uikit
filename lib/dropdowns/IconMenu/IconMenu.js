'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _More = require('../../icons/More');

var _More2 = _interopRequireDefault(_More);

var _Dropdown = require('../Dropdown');

var _Dropdown2 = _interopRequireDefault(_Dropdown);

var _Option = require('../Option');

var _Option2 = _interopRequireDefault(_Option);

var _IconMenu = require('./IconMenu.scss');

var _IconMenu2 = _interopRequireDefault(_IconMenu);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var IconMenu = function IconMenu(_ref) {
  var icon = _ref.icon,
      options = _ref.options,
      children = _ref.children,
      rest = _objectWithoutProperties(_ref, ['icon', 'options', 'children']);

  return _react2.default.createElement(
    _Dropdown2.default,
    _extends({
      trigger: _react2.default.createElement(
        'div',
        { className: _IconMenu2.default.icon },
        icon
      )
    }, rest),
    options ? options.map(function (_ref2, index) {
      var value = _ref2.value,
          optionRest = _objectWithoutProperties(_ref2, ['value']);

      return _react2.default.createElement(
        _Option2.default,
        _extends({ key: index }, optionRest),
        value
      );
    }) : children
  );
};

IconMenu.propTypes = {
  icon: _propTypes2.default.node,
  children: _propTypes2.default.node,
  options: _propTypes2.default.array
};

IconMenu.defaultProps = {
  icon: _react2.default.createElement(_More2.default, null)
};

exports.default = IconMenu;