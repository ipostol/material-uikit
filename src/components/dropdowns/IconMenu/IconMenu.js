import React from 'react';
import PropTypes from 'prop-types';
import DropdownIcon from '../../icons/More';
import Dropdown from '../Dropdown';
import Option from '../Option';
import styles from './IconMenu.scss';

const IconMenu = ({ icon, options, children, ...rest }) => (
  <Dropdown
    trigger={<div className={styles.icon}>{icon}</div>}
    {...rest}
  >
    {
      options ? options.map(({ value, ...optionRest }, index) => (
        <Option key={index} {...optionRest}>
          {value}
        </Option>
      )) : children
    }
  </Dropdown>
);

IconMenu.propTypes = {
  icon: PropTypes.node,
  children: PropTypes.node,
  options: PropTypes.array,
};

IconMenu.defaultProps = {
  icon: <DropdownIcon />,
};

export default IconMenu;
