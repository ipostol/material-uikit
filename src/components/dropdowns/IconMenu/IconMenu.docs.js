import React from 'react';
import { Skin } from 'react-markup-components';
import DropdownIcon from './../../icons/More';
import IconMenu from './IconMenu';

const options = [{
  value: 'Option1',
}, {
  value: 'Option2',
}];

export const IconMenuDocs = () => (
  <Skin>
    <IconMenu
      options={options}
      onRight
      icon={<DropdownIcon />}
    />
    <br />
    <br />
  </Skin>
);

export default IconMenuDocs;
