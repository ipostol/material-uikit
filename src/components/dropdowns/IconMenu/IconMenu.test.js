import React from 'react';
import { shallow } from 'enzyme';
import IconMenu from './IconMenu';

describe('IconMenu component', () => {

  it('should render correctly', () => {

    const props = {
      options: [{ id: 1 }],
    };

    const wrapper = shallow(
      <IconMenu {...props} />,
    );

    expect(wrapper).toMatchSnapshot();

  });

  it('should render correctly children props', () => {

    const props = {
      children: <div />,
    };

    const wrapper = shallow(
      <IconMenu {...props} />,
    );

    expect(wrapper).toMatchSnapshot();

  });

});
