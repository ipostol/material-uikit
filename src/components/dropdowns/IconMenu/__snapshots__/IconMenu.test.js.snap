// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconMenu component should render correctly 1`] = `
<Dropdown
  trigger={
    <div
      className="icon"
    >
      <More />
    </div>
  }
>
  <Option
    element="div"
    id={1}
    size="s"
  />
</Dropdown>
`;

exports[`IconMenu component should render correctly children props 1`] = `
<Dropdown
  trigger={
    <div
      className="icon"
    >
      <More />
    </div>
  }
>
  <div />
</Dropdown>
`;
